src/utils/3dModelFileContext.jsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useState } from "react";
+import PropTypes from "prop-types";
+
+export const ModelFileContext = createContext();
+
+export const ModelFileProvider = ({ children }) => {
+  const [fileFor3dModel, setFileFor3dModel] = useState(null);
+
+  return (
+    <ModelFileContext.Provider value={{ fileFor3dModel, setFileFor3dModel }}>
+      {children}
+    </ModelFileContext.Provider>
+  );
+};
+
+ModelFileProvider.propTypes = {
+  children: PropTypes.node,
+};
```

Comment:
I think that using Context for the file data is a bit overkill. From what I understand, context is usually used for things that you want to be able to be used globally by all of your components but the file data is only really used by stuff in the AnalysisPage so I think the variable holding the file data would be better as state in the AnalysisPage component and then it can be passed into the model viewer component as a prop. And then I think the setter function would be passed into the upload button as a prop too. I'm not too experienced with this stuff though so let me know if theres a reason why you prefer using a context in this case.

Also you can ask me any questions here or in a zoom meeting if you want more details about state/props vs context

Reply:
Yeah you are absolutely correct, I think it may have just gotten late and i remembered from the bootcamp that you're not supposed to pass props upwards. I confused myself and just ended up creating a context file to take care of this.

I updated the state to be defined in the analysis page file!